api/backend/ecs/environment_manager.go

```diff
@@ -327,6 +328,102 @@ func (this *ECSEnvironmentManager) DeleteEnvironment(environmentID string) error
 	return nil
 }
 
+func (this *ECSEnvironmentManager) CreateEnvironmentLink(sourceEnvironmentID, destEnvironmentID string) error {
+	sourceECSID := id.L0EnvironmentID(sourceEnvironmentID).ECSEnvironmentID()
+	destECSID := id.L0EnvironmentID(destEnvironmentID).ECSEnvironmentID()
+
+	sourceGroup, err := this.getEnvironmentSecurityGroup(sourceECSID)
+	if err != nil {
+		return err
+	}
+
+	destGroup, err := this.getEnvironmentSecurityGroup(destECSID)
+	if err != nil {
+		return err
+	}
+
+	if err := this.EC2.AuthorizeSecurityGroupIngressFromGroup(sourceGroup.GroupId, destGroup.GroupId); err != nil {
+		if !ContainsErrCode(err, "InvalidPermission.Duplicate") {
+			return err
+		}
+	}
+
+	if err := this.EC2.AuthorizeSecurityGroupIngressFromGroup(destGroup.GroupId, sourceGroup.GroupId); err != nil {
+		if !ContainsErrCode(err, "InvalidPermission.Duplicate") {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (this *ECSEnvironmentManager) DeleteEnvironmentLink(sourceEnvironmentID, destEnvironmentID string) error {
+	sourceECSID := id.L0EnvironmentID(sourceEnvironmentID).ECSEnvironmentID()
+	destECSID := id.L0EnvironmentID(destEnvironmentID).ECSEnvironmentID()
+
+	sourceGroup, err := this.EC2.DescribeSecurityGroup(sourceECSID.SecurityGroupName())
+	if err != nil {
+		return err
+	}
+
+	if sourceGroup == nil {
+		log.Warnf("Skipping environment unlink since security group '%s' does not exist", sourceECSID.SecurityGroupName())
```

Comment:
Maybe a philosophical question, but wouldn't this be an error? Most delete operations I can think of return an error if the thing being deleted doesn't exist.